store/store.go

```diff
@@ -1445,8 +1459,12 @@ var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo,
 
 	partialTargetPath := targetPath + ".partial"
 
+	xdelta3Bin, err := getXdelta3Binary()
+	if err != nil {
+		return err
+	}
 	xdelta3Args := []string{"-d", "-s", snapPath, deltaPath, partialTargetPath}
-	cmd := exec.Command("xdelta3", xdelta3Args...)
+	cmd := exec.Command(xdelta3Bin, xdelta3Args...)
```

Comment:
Thanks @zyga - I implemented a osutil.CommandFromCore() but it got a bit out of hand. We may need to split this branch.